// asposewordscloud/dist/model/availableFontsResponse.js
"use strict";
/*
 * --------------------------------------------------------------------------------
 * <copyright company="Aspose" file="availableFontsResponse.ts">
 *   Copyright (c) 2024 Aspose.Words for Cloud
 * </copyright>
 * <summary>
 *   Permission is hereby granted, free of charge, to any person obtaining a copy
 *  of this software and associated documentation files (the "Software"), to deal
 *  in the Software without restriction, including without limitation the rights
 *  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 *  copies of the Software, and to permit persons to whom the Software is
 *  furnished to do so, subject to the following conditions:
 *
 *  The above copyright notice and this permission notice shall be included in all
 *  copies or substantial portions of the Software.
 *
 *  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 *  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 *  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 *  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 *  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 *  SOFTWARE.
 * </summary>
 * --------------------------------------------------------------------------------
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AvailableFontsResponse = exports.importsMapAvailableFontsResponse = void 0;
const fontInfo_1 = require("./fontInfo");
const wordsResponse_1 = require("./wordsResponse");
exports.importsMapAvailableFontsResponse = {
    FontInfo: fontInfo_1.FontInfo,
    WordsResponse: wordsResponse_1.WordsResponse,
};
/**
 * The REST response with data on system, additional and custom fonts, available for document processing.
 */
class AvailableFontsResponse extends wordsResponse_1.WordsResponse {
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return super.getAttributeTypeMap().concat(AvailableFontsResponse.attributeTypeMap);
    }
    constructor(init) {
        super(init);
        Object.assign(this, init);
    }
    collectFilesContent(_resultFilesContent) {
    }
    validate() {
        super.validate();
        if (this.additionalFonts !== null && this.additionalFonts !== undefined) {
            for (let elementAdditionalFonts of this.additionalFonts) {
                elementAdditionalFonts === null || elementAdditionalFonts === void 0 ? void 0 : elementAdditionalFonts.validate();
            }
        }
        if (this.customFonts !== null && this.customFonts !== undefined) {
            for (let elementCustomFonts of this.customFonts) {
                elementCustomFonts === null || elementCustomFonts === void 0 ? void 0 : elementCustomFonts.validate();
            }
        }
        if (this.systemFonts !== null && this.systemFonts !== undefined) {
            for (let elementSystemFonts of this.systemFonts) {
                elementSystemFonts === null || elementSystemFonts === void 0 ? void 0 : elementSystemFonts.validate();
            }
        }
    }
}
exports.AvailableFontsResponse = AvailableFontsResponse;
/**
 * Attribute type map
 */
AvailableFontsResponse.attributeTypeMap = [
    {
        name: "additionalFonts",
        baseName: "AdditionalFonts",
        type: "Array<FontInfo>",
    },
    {
        name: "customFonts",
        baseName: "CustomFonts",
        type: "Array<FontInfo>",
    },
    {
        name: "systemFonts",
        baseName: "SystemFonts",
        type: "Array<FontInfo>",
    }
];
