// asposewordscloud/dist/model/baseEntry.js
"use strict";
/*
 * --------------------------------------------------------------------------------
 * <copyright company="Aspose" file="baseEntry.ts">
 *   Copyright (c) 2024 Aspose.Words for Cloud
 * </copyright>
 * <summary>
 *   Permission is hereby granted, free of charge, to any person obtaining a copy
 *  of this software and associated documentation files (the "Software"), to deal
 *  in the Software without restriction, including without limitation the rights
 *  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 *  copies of the Software, and to permit persons to whom the Software is
 *  furnished to do so, subject to the following conditions:
 *
 *  The above copyright notice and this permission notice shall be included in all
 *  copies or substantial portions of the Software.
 *
 *  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 *  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 *  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 *  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 *  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 *  SOFTWARE.
 * </summary>
 * --------------------------------------------------------------------------------
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.BaseEntry = exports.importsMapBaseEntry = void 0;
const fileReference_1 = require("./fileReference");
exports.importsMapBaseEntry = {
    FileReference: fileReference_1.FileReference,
};
/**
 * Represents a base class for document which will be appended to the original resource document.
 */
class BaseEntry {
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return BaseEntry.attributeTypeMap;
    }
    constructor(init) {
        Object.assign(this, init);
    }
    collectFilesContent(_resultFilesContent) {
        if (this.fileReference) {
            this.fileReference.collectFilesContent(_resultFilesContent);
        }
    }
    validate() {
        var _a;
        if (this.fileReference === null || this.fileReference === undefined) {
            throw new Error('Property FileReference in BaseEntry is required.');
        }
        (_a = this.fileReference) === null || _a === void 0 ? void 0 : _a.validate();
    }
}
exports.BaseEntry = BaseEntry;
/**
 * Attribute type map
 */
BaseEntry.attributeTypeMap = [
    {
        name: "fileReference",
        baseName: "FileReference",
        type: "FileReference",
    }
];
