// asposewordscloud/dist/model/bookmarks.js
"use strict";
/*
 * --------------------------------------------------------------------------------
 * <copyright company="Aspose" file="bookmarks.ts">
 *   Copyright (c) 2024 Aspose.Words for Cloud
 * </copyright>
 * <summary>
 *   Permission is hereby granted, free of charge, to any person obtaining a copy
 *  of this software and associated documentation files (the "Software"), to deal
 *  in the Software without restriction, including without limitation the rights
 *  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 *  copies of the Software, and to permit persons to whom the Software is
 *  furnished to do so, subject to the following conditions:
 *
 *  The above copyright notice and this permission notice shall be included in all
 *  copies or substantial portions of the Software.
 *
 *  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 *  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 *  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 *  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 *  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 *  SOFTWARE.
 * </summary>
 * --------------------------------------------------------------------------------
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Bookmarks = exports.importsMapBookmarks = void 0;
const bookmark_1 = require("./bookmark");
const linkElement_1 = require("./linkElement");
exports.importsMapBookmarks = {
    Bookmark: bookmark_1.Bookmark,
    LinkElement: linkElement_1.LinkElement,
};
/**
 * Represents an array of bookmarks.
 */
class Bookmarks extends linkElement_1.LinkElement {
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return super.getAttributeTypeMap().concat(Bookmarks.attributeTypeMap);
    }
    constructor(init) {
        super(init);
        Object.assign(this, init);
    }
    collectFilesContent(_resultFilesContent) {
    }
    validate() {
        super.validate();
        if (this.bookmarkList !== null && this.bookmarkList !== undefined) {
            for (let elementBookmarkList of this.bookmarkList) {
                elementBookmarkList === null || elementBookmarkList === void 0 ? void 0 : elementBookmarkList.validate();
            }
        }
    }
}
exports.Bookmarks = Bookmarks;
/**
 * Attribute type map
 */
Bookmarks.attributeTypeMap = [
    {
        name: "bookmarkList",
        baseName: "BookmarkList",
        type: "Array<Bookmark>",
    }
];
