// asposewordscloud/dist/model/classificationResponse.js
"use strict";
/*
 * --------------------------------------------------------------------------------
 * <copyright company="Aspose" file="classificationResponse.ts">
 *   Copyright (c) 2024 Aspose.Words for Cloud
 * </copyright>
 * <summary>
 *   Permission is hereby granted, free of charge, to any person obtaining a copy
 *  of this software and associated documentation files (the "Software"), to deal
 *  in the Software without restriction, including without limitation the rights
 *  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 *  copies of the Software, and to permit persons to whom the Software is
 *  furnished to do so, subject to the following conditions:
 *
 *  The above copyright notice and this permission notice shall be included in all
 *  copies or substantial portions of the Software.
 *
 *  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 *  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 *  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 *  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 *  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 *  SOFTWARE.
 * </summary>
 * --------------------------------------------------------------------------------
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ClassificationResponse = exports.importsMapClassificationResponse = void 0;
const classificationResult_1 = require("./classificationResult");
const wordsResponse_1 = require("./wordsResponse");
exports.importsMapClassificationResponse = {
    ClassificationResult: classificationResult_1.ClassificationResult,
    WordsResponse: wordsResponse_1.WordsResponse,
};
/**
 * The REST response with data on multi-class text classification.
 * This response is returned by the Service when handling "PUT https://api.aspose.cloud/v4.0/words/classify" REST API requests.
 */
class ClassificationResponse extends wordsResponse_1.WordsResponse {
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return super.getAttributeTypeMap().concat(ClassificationResponse.attributeTypeMap);
    }
    constructor(init) {
        super(init);
        Object.assign(this, init);
    }
    collectFilesContent(_resultFilesContent) {
    }
    validate() {
        super.validate();
        if (this.bestClassProbability === null || this.bestClassProbability === undefined) {
            throw new Error('Property BestClassProbability in ClassificationResponse is required.');
        }
        if (this.bestResults !== null && this.bestResults !== undefined) {
            for (let elementBestResults of this.bestResults) {
                elementBestResults === null || elementBestResults === void 0 ? void 0 : elementBestResults.validate();
            }
        }
    }
}
exports.ClassificationResponse = ClassificationResponse;
/**
 * Attribute type map
 */
ClassificationResponse.attributeTypeMap = [
    {
        name: "bestClassName",
        baseName: "BestClassName",
        type: "string",
    },
    {
        name: "bestClassProbability",
        baseName: "BestClassProbability",
        type: "number",
    },
    {
        name: "bestResults",
        baseName: "BestResults",
        type: "Array<ClassificationResult>",
    }
];
