// asposewordscloud/dist/model/commentsCollection.js
"use strict";
/*
 * --------------------------------------------------------------------------------
 * <copyright company="Aspose" file="commentsCollection.ts">
 *   Copyright (c) 2024 Aspose.Words for Cloud
 * </copyright>
 * <summary>
 *   Permission is hereby granted, free of charge, to any person obtaining a copy
 *  of this software and associated documentation files (the "Software"), to deal
 *  in the Software without restriction, including without limitation the rights
 *  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 *  copies of the Software, and to permit persons to whom the Software is
 *  furnished to do so, subject to the following conditions:
 *
 *  The above copyright notice and this permission notice shall be included in all
 *  copies or substantial portions of the Software.
 *
 *  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 *  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 *  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 *  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 *  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 *  SOFTWARE.
 * </summary>
 * --------------------------------------------------------------------------------
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CommentsCollection = exports.importsMapCommentsCollection = void 0;
const comment_1 = require("./comment");
const linkElement_1 = require("./linkElement");
exports.importsMapCommentsCollection = {
    Comment: comment_1.Comment,
    LinkElement: linkElement_1.LinkElement,
};
/**
 * The collection of comments.
 */
class CommentsCollection extends linkElement_1.LinkElement {
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return super.getAttributeTypeMap().concat(CommentsCollection.attributeTypeMap);
    }
    constructor(init) {
        super(init);
        Object.assign(this, init);
    }
    collectFilesContent(_resultFilesContent) {
    }
    validate() {
        super.validate();
        if (this.commentList !== null && this.commentList !== undefined) {
            for (let elementCommentList of this.commentList) {
                elementCommentList === null || elementCommentList === void 0 ? void 0 : elementCommentList.validate();
            }
        }
    }
}
exports.CommentsCollection = CommentsCollection;
/**
 * Attribute type map
 */
CommentsCollection.attributeTypeMap = [
    {
        name: "commentList",
        baseName: "CommentList",
        type: "Array<Comment>",
    }
];
