// asposewordscloud/dist/model/commentBase.js
"use strict";
/*
 * --------------------------------------------------------------------------------
 * <copyright company="Aspose" file="commentBase.ts">
 *   Copyright (c) 2024 Aspose.Words for Cloud
 * </copyright>
 * <summary>
 *   Permission is hereby granted, free of charge, to any person obtaining a copy
 *  of this software and associated documentation files (the "Software"), to deal
 *  in the Software without restriction, including without limitation the rights
 *  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 *  copies of the Software, and to permit persons to whom the Software is
 *  furnished to do so, subject to the following conditions:
 *
 *  The above copyright notice and this permission notice shall be included in all
 *  copies or substantial portions of the Software.
 *
 *  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 *  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 *  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 *  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 *  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 *  SOFTWARE.
 * </summary>
 * --------------------------------------------------------------------------------
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CommentBase = exports.importsMapCommentBase = void 0;
const positionInsideNode_1 = require("./positionInsideNode");
exports.importsMapCommentBase = {
    PositionInsideNode: positionInsideNode_1.PositionInsideNode,
};
/**
 * Comment.
 */
class CommentBase {
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return CommentBase.attributeTypeMap;
    }
    constructor(init) {
        Object.assign(this, init);
    }
    collectFilesContent(_resultFilesContent) {
    }
    validate() {
        var _a, _b;
        if (this.rangeStart === null || this.rangeStart === undefined) {
            throw new Error('Property RangeStart in CommentBase is required.');
        }
        if (this.rangeEnd === null || this.rangeEnd === undefined) {
            throw new Error('Property RangeEnd in CommentBase is required.');
        }
        if (this.author === null || this.author === undefined) {
            throw new Error('Property Author in CommentBase is required.');
        }
        if (this.initial === null || this.initial === undefined) {
            throw new Error('Property Initial in CommentBase is required.');
        }
        if (this.text === null || this.text === undefined) {
            throw new Error('Property Text in CommentBase is required.');
        }
        (_a = this.rangeStart) === null || _a === void 0 ? void 0 : _a.validate();
        (_b = this.rangeEnd) === null || _b === void 0 ? void 0 : _b.validate();
    }
}
exports.CommentBase = CommentBase;
/**
 * Attribute type map
 */
CommentBase.attributeTypeMap = [
    {
        name: "rangeStart",
        baseName: "RangeStart",
        type: "PositionInsideNode",
    },
    {
        name: "rangeEnd",
        baseName: "RangeEnd",
        type: "PositionInsideNode",
    },
    {
        name: "author",
        baseName: "Author",
        type: "string",
    },
    {
        name: "initial",
        baseName: "Initial",
        type: "string",
    },
    {
        name: "dateTime",
        baseName: "DateTime",
        type: "Date",
    },
    {
        name: "text",
        baseName: "Text",
        type: "string",
    }
];
