// asposewordscloud/dist/model/customXmlPart.js
"use strict";
/*
 * --------------------------------------------------------------------------------
 * <copyright company="Aspose" file="customXmlPart.ts">
 *   Copyright (c) 2024 Aspose.Words for Cloud
 * </copyright>
 * <summary>
 *   Permission is hereby granted, free of charge, to any person obtaining a copy
 *  of this software and associated documentation files (the "Software"), to deal
 *  in the Software without restriction, including without limitation the rights
 *  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 *  copies of the Software, and to permit persons to whom the Software is
 *  furnished to do so, subject to the following conditions:
 *
 *  The above copyright notice and this permission notice shall be included in all
 *  copies or substantial portions of the Software.
 *
 *  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 *  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 *  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 *  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 *  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 *  SOFTWARE.
 * </summary>
 * --------------------------------------------------------------------------------
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomXmlPart = exports.importsMapCustomXmlPart = void 0;
const customXmlPartLink_1 = require("./customXmlPartLink");
exports.importsMapCustomXmlPart = {
    CustomXmlPartLink: customXmlPartLink_1.CustomXmlPartLink,
};
/**
 * DTO container with a CustomXmlPart.
 */
class CustomXmlPart extends customXmlPartLink_1.CustomXmlPartLink {
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return super.getAttributeTypeMap().concat(CustomXmlPart.attributeTypeMap);
    }
    constructor(init) {
        super(init);
        Object.assign(this, init);
    }
    collectFilesContent(_resultFilesContent) {
    }
    validate() {
        super.validate();
        if (this.data === null || this.data === undefined) {
            throw new Error('Property Data in CustomXmlPart is required.');
        }
    }
}
exports.CustomXmlPart = CustomXmlPart;
/**
 * Attribute type map
 */
CustomXmlPart.attributeTypeMap = [
    {
        name: "id",
        baseName: "Id",
        type: "string",
    },
    {
        name: "data",
        baseName: "Data",
        type: "string",
    }
];
