// asposewordscloud/dist/model/filesUploadResult.js
"use strict";
/*
 * --------------------------------------------------------------------------------
 * <copyright company="Aspose" file="filesUploadResult.ts">
 *   Copyright (c) 2024 Aspose.Words for Cloud
 * </copyright>
 * <summary>
 *   Permission is hereby granted, free of charge, to any person obtaining a copy
 *  of this software and associated documentation files (the "Software"), to deal
 *  in the Software without restriction, including without limitation the rights
 *  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 *  copies of the Software, and to permit persons to whom the Software is
 *  furnished to do so, subject to the following conditions:
 *
 *  The above copyright notice and this permission notice shall be included in all
 *  copies or substantial portions of the Software.
 *
 *  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 *  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 *  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 *  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 *  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 *  SOFTWARE.
 * </summary>
 * --------------------------------------------------------------------------------
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.FilesUploadResult = exports.importsMapFilesUploadResult = void 0;
const error_1 = require("./error");
exports.importsMapFilesUploadResult = {
    Error: error_1.Error,
};
/**
 * File upload result.
 */
class FilesUploadResult {
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return FilesUploadResult.attributeTypeMap;
    }
    constructor(init) {
        Object.assign(this, init);
    }
    collectFilesContent(_resultFilesContent) {
    }
    validate() {
        if (this.errors !== null && this.errors !== undefined) {
            for (let elementErrors of this.errors) {
                elementErrors === null || elementErrors === void 0 ? void 0 : elementErrors.validate();
            }
        }
    }
}
exports.FilesUploadResult = FilesUploadResult;
/**
 * Attribute type map
 */
FilesUploadResult.attributeTypeMap = [
    {
        name: "errors",
        baseName: "Errors",
        type: "Array<Error>",
    },
    {
        name: "uploaded",
        baseName: "Uploaded",
        type: "Array<string>",
    }
];
