// asposewordscloud/dist/model/flatOpcSaveOptionsData.js
"use strict";
/*
 * --------------------------------------------------------------------------------
 * <copyright company="Aspose" file="flatOpcSaveOptionsData.ts">
 *   Copyright (c) 2024 Aspose.Words for Cloud
 * </copyright>
 * <summary>
 *   Permission is hereby granted, free of charge, to any person obtaining a copy
 *  of this software and associated documentation files (the "Software"), to deal
 *  in the Software without restriction, including without limitation the rights
 *  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 *  copies of the Software, and to permit persons to whom the Software is
 *  furnished to do so, subject to the following conditions:
 *
 *  The above copyright notice and this permission notice shall be included in all
 *  copies or substantial portions of the Software.
 *
 *  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 *  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 *  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 *  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 *  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 *  SOFTWARE.
 * </summary>
 * --------------------------------------------------------------------------------
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.FlatOpcSaveOptionsData = exports.importsMapFlatOpcSaveOptionsData = void 0;
const ooxmlSaveOptionsData_1 = require("./ooxmlSaveOptionsData");
exports.importsMapFlatOpcSaveOptionsData = {
    OoxmlSaveOptionsData: ooxmlSaveOptionsData_1.OoxmlSaveOptionsData,
};
/**
 * Container class for fopc save options.
 */
class FlatOpcSaveOptionsData extends ooxmlSaveOptionsData_1.OoxmlSaveOptionsData {
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return super.getAttributeTypeMap().concat(FlatOpcSaveOptionsData.attributeTypeMap);
    }
    constructor(init) {
        super(init);
        this.saveFormat = 'fopc';
        Object.assign(this, init);
    }
    collectFilesContent(_resultFilesContent) {
    }
    validate() {
        super.validate();
    }
}
exports.FlatOpcSaveOptionsData = FlatOpcSaveOptionsData;
/**
 * Attribute type map
 */
FlatOpcSaveOptionsData.attributeTypeMap = [];
