// asposewordscloud/dist/model/formFieldCheckbox.js
"use strict";
/*
 * --------------------------------------------------------------------------------
 * <copyright company="Aspose" file="formFieldCheckbox.ts">
 *   Copyright (c) 2024 Aspose.Words for Cloud
 * </copyright>
 * <summary>
 *   Permission is hereby granted, free of charge, to any person obtaining a copy
 *  of this software and associated documentation files (the "Software"), to deal
 *  in the Software without restriction, including without limitation the rights
 *  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 *  copies of the Software, and to permit persons to whom the Software is
 *  furnished to do so, subject to the following conditions:
 *
 *  The above copyright notice and this permission notice shall be included in all
 *  copies or substantial portions of the Software.
 *
 *  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 *  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 *  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 *  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 *  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 *  SOFTWARE.
 * </summary>
 * --------------------------------------------------------------------------------
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.FormFieldCheckbox = exports.importsMapFormFieldCheckbox = void 0;
const formField_1 = require("./formField");
exports.importsMapFormFieldCheckbox = {
    FormField: formField_1.FormField,
};
/**
 * FormField checkbox element.
 */
class FormFieldCheckbox extends formField_1.FormField {
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return super.getAttributeTypeMap().concat(FormFieldCheckbox.attributeTypeMap);
    }
    constructor(init) {
        super(init);
        Object.assign(this, init);
    }
    collectFilesContent(_resultFilesContent) {
    }
    validate() {
        super.validate();
        if (this.checked === null || this.checked === undefined) {
            throw new Error('Property Checked in FormFieldCheckbox is required.');
        }
    }
}
exports.FormFieldCheckbox = FormFieldCheckbox;
/**
 * Attribute type map
 */
FormFieldCheckbox.attributeTypeMap = [
    {
        name: "isCheckBoxExactSize",
        baseName: "IsCheckBoxExactSize",
        type: "boolean",
    },
    {
        name: "checkBoxSize",
        baseName: "CheckBoxSize",
        type: "number",
    },
    {
        name: "checked",
        baseName: "Checked",
        type: "boolean",
    }
];
