// asposewordscloud/dist/model/formFieldCollection.js
"use strict";
/*
 * --------------------------------------------------------------------------------
 * <copyright company="Aspose" file="formFieldCollection.ts">
 *   Copyright (c) 2024 Aspose.Words for Cloud
 * </copyright>
 * <summary>
 *   Permission is hereby granted, free of charge, to any person obtaining a copy
 *  of this software and associated documentation files (the "Software"), to deal
 *  in the Software without restriction, including without limitation the rights
 *  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 *  copies of the Software, and to permit persons to whom the Software is
 *  furnished to do so, subject to the following conditions:
 *
 *  The above copyright notice and this permission notice shall be included in all
 *  copies or substantial portions of the Software.
 *
 *  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 *  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 *  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 *  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 *  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 *  SOFTWARE.
 * </summary>
 * --------------------------------------------------------------------------------
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.FormFieldCollection = exports.importsMapFormFieldCollection = void 0;
const formField_1 = require("./formField");
const linkElement_1 = require("./linkElement");
exports.importsMapFormFieldCollection = {
    FormField: formField_1.FormField,
    LinkElement: linkElement_1.LinkElement,
};
/**
 * DTO container with a collection of form fields.
 */
class FormFieldCollection extends linkElement_1.LinkElement {
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return super.getAttributeTypeMap().concat(FormFieldCollection.attributeTypeMap);
    }
    constructor(init) {
        super(init);
        Object.assign(this, init);
    }
    collectFilesContent(_resultFilesContent) {
    }
    validate() {
        super.validate();
        if (this.list !== null && this.list !== undefined) {
            for (let elementList of this.list) {
                elementList === null || elementList === void 0 ? void 0 : elementList.validate();
            }
        }
    }
}
exports.FormFieldCollection = FormFieldCollection;
/**
 * Attribute type map
 */
FormFieldCollection.attributeTypeMap = [
    {
        name: "list",
        baseName: "List",
        type: "Array<FormField>",
    }
];
