// asposewordscloud/dist/model/formField.js
"use strict";
/*
 * --------------------------------------------------------------------------------
 * <copyright company="Aspose" file="formField.ts">
 *   Copyright (c) 2024 Aspose.Words for Cloud
 * </copyright>
 * <summary>
 *   Permission is hereby granted, free of charge, to any person obtaining a copy
 *  of this software and associated documentation files (the "Software"), to deal
 *  in the Software without restriction, including without limitation the rights
 *  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 *  copies of the Software, and to permit persons to whom the Software is
 *  furnished to do so, subject to the following conditions:
 *
 *  The above copyright notice and this permission notice shall be included in all
 *  copies or substantial portions of the Software.
 *
 *  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 *  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 *  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 *  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 *  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 *  SOFTWARE.
 * </summary>
 * --------------------------------------------------------------------------------
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.FormField = exports.importsMapFormField = void 0;
const nodeLink_1 = require("./nodeLink");
exports.importsMapFormField = {
    NodeLink: nodeLink_1.NodeLink,
};
/**
 * FromField.
 */
class FormField extends nodeLink_1.NodeLink {
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return super.getAttributeTypeMap().concat(FormField.attributeTypeMap);
    }
    constructor(init) {
        super(init);
        Object.assign(this, init);
    }
    collectFilesContent(_resultFilesContent) {
    }
    validate() {
        super.validate();
        if (this.name === null || this.name === undefined) {
            throw new Error('Property Name in FormField is required.');
        }
    }
}
exports.FormField = FormField;
/**
 * Attribute type map
 */
FormField.attributeTypeMap = [
    {
        name: "name",
        baseName: "Name",
        type: "string",
    },
    {
        name: "enabled",
        baseName: "Enabled",
        type: "boolean",
    },
    {
        name: "statusText",
        baseName: "StatusText",
        type: "string",
    },
    {
        name: "ownStatus",
        baseName: "OwnStatus",
        type: "boolean",
    },
    {
        name: "helpText",
        baseName: "HelpText",
        type: "string",
    },
    {
        name: "ownHelp",
        baseName: "OwnHelp",
        type: "boolean",
    },
    {
        name: "calculateOnExit",
        baseName: "CalculateOnExit",
        type: "boolean",
    },
    {
        name: "entryMacro",
        baseName: "EntryMacro",
        type: "string",
    },
    {
        name: "exitMacro",
        baseName: "ExitMacro",
        type: "string",
    }
];
