// asposewordscloud/dist/model/headerFooterLink.js
"use strict";
/*
 * --------------------------------------------------------------------------------
 * <copyright company="Aspose" file="headerFooterLink.ts">
 *   Copyright (c) 2024 Aspose.Words for Cloud
 * </copyright>
 * <summary>
 *   Permission is hereby granted, free of charge, to any person obtaining a copy
 *  of this software and associated documentation files (the "Software"), to deal
 *  in the Software without restriction, including without limitation the rights
 *  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 *  copies of the Software, and to permit persons to whom the Software is
 *  furnished to do so, subject to the following conditions:
 *
 *  The above copyright notice and this permission notice shall be included in all
 *  copies or substantial portions of the Software.
 *
 *  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 *  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 *  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 *  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 *  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 *  SOFTWARE.
 * </summary>
 * --------------------------------------------------------------------------------
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.HeaderFooterLink = exports.importsMapHeaderFooterLink = void 0;
const linkElement_1 = require("./linkElement");
exports.importsMapHeaderFooterLink = {
    LinkElement: linkElement_1.LinkElement,
};
/**
 * HeaderFooter link element.
 */
class HeaderFooterLink extends linkElement_1.LinkElement {
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return super.getAttributeTypeMap().concat(HeaderFooterLink.attributeTypeMap);
    }
    constructor(init) {
        super(init);
        Object.assign(this, init);
    }
    collectFilesContent(_resultFilesContent) {
    }
    validate() {
        super.validate();
        if (this.type === null || this.type === undefined) {
            throw new Error('Property Type in HeaderFooterLink is required.');
        }
    }
}
exports.HeaderFooterLink = HeaderFooterLink;
/**
 * Attribute type map
 */
HeaderFooterLink.attributeTypeMap = [
    {
        name: "type",
        baseName: "Type",
        type: "HeaderFooterLink.TypeEnum",
    }
];
/**
 * Enums for HeaderFooterLink
 */
// tslint:disable:quotemark
// tslint:disable-next-line:no-namespace
(function (HeaderFooterLink) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum[TypeEnum["HeaderEven"] = 'HeaderEven'] = "HeaderEven";
        TypeEnum[TypeEnum["HeaderPrimary"] = 'HeaderPrimary'] = "HeaderPrimary";
        TypeEnum[TypeEnum["FooterEven"] = 'FooterEven'] = "FooterEven";
        TypeEnum[TypeEnum["FooterPrimary"] = 'FooterPrimary'] = "FooterPrimary";
        TypeEnum[TypeEnum["HeaderFirst"] = 'HeaderFirst'] = "HeaderFirst";
        TypeEnum[TypeEnum["FooterFirst"] = 'FooterFirst'] = "FooterFirst";
    })(TypeEnum = HeaderFooterLink.TypeEnum || (HeaderFooterLink.TypeEnum = {}));
})(HeaderFooterLink = exports.HeaderFooterLink || (exports.HeaderFooterLink = {}));
// tslint:enable:quotemark
