// asposewordscloud/dist/model/headerFooter.js
"use strict";
/*
 * --------------------------------------------------------------------------------
 * <copyright company="Aspose" file="headerFooter.ts">
 *   Copyright (c) 2024 Aspose.Words for Cloud
 * </copyright>
 * <summary>
 *   Permission is hereby granted, free of charge, to any person obtaining a copy
 *  of this software and associated documentation files (the "Software"), to deal
 *  in the Software without restriction, including without limitation the rights
 *  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 *  copies of the Software, and to permit persons to whom the Software is
 *  furnished to do so, subject to the following conditions:
 *
 *  The above copyright notice and this permission notice shall be included in all
 *  copies or substantial portions of the Software.
 *
 *  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 *  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 *  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 *  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 *  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 *  SOFTWARE.
 * </summary>
 * --------------------------------------------------------------------------------
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.HeaderFooter = exports.importsMapHeaderFooter = void 0;
const headerFooterLink_1 = require("./headerFooterLink");
const linkElement_1 = require("./linkElement");
const nodeLink_1 = require("./nodeLink");
exports.importsMapHeaderFooter = {
    HeaderFooterLink: headerFooterLink_1.HeaderFooterLink,
    LinkElement: linkElement_1.LinkElement,
    NodeLink: nodeLink_1.NodeLink,
};
/**
 * DTO container with a section element.
 */
class HeaderFooter extends headerFooterLink_1.HeaderFooterLink {
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return super.getAttributeTypeMap().concat(HeaderFooter.attributeTypeMap);
    }
    constructor(init) {
        super(init);
        Object.assign(this, init);
    }
    collectFilesContent(_resultFilesContent) {
    }
    validate() {
        var _a, _b;
        super.validate();
        if (this.childNodes !== null && this.childNodes !== undefined) {
            for (let elementChildNodes of this.childNodes) {
                elementChildNodes === null || elementChildNodes === void 0 ? void 0 : elementChildNodes.validate();
            }
        }
        (_a = this.paragraphs) === null || _a === void 0 ? void 0 : _a.validate();
        (_b = this.drawingObjects) === null || _b === void 0 ? void 0 : _b.validate();
    }
}
exports.HeaderFooter = HeaderFooter;
/**
 * Attribute type map
 */
HeaderFooter.attributeTypeMap = [
    {
        name: "childNodes",
        baseName: "ChildNodes",
        type: "Array<NodeLink>",
    },
    {
        name: "paragraphs",
        baseName: "Paragraphs",
        type: "LinkElement",
    },
    {
        name: "drawingObjects",
        baseName: "DrawingObjects",
        type: "LinkElement",
    }
];
