// asposewordscloud/dist/model/imageEntryList.js
"use strict";
/*
 * --------------------------------------------------------------------------------
 * <copyright company="Aspose" file="imageEntryList.ts">
 *   Copyright (c) 2024 Aspose.Words for Cloud
 * </copyright>
 * <summary>
 *   Permission is hereby granted, free of charge, to any person obtaining a copy
 *  of this software and associated documentation files (the "Software"), to deal
 *  in the Software without restriction, including without limitation the rights
 *  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 *  copies of the Software, and to permit persons to whom the Software is
 *  furnished to do so, subject to the following conditions:
 *
 *  The above copyright notice and this permission notice shall be included in all
 *  copies or substantial portions of the Software.
 *
 *  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 *  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 *  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 *  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 *  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 *  SOFTWARE.
 * </summary>
 * --------------------------------------------------------------------------------
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ImageEntryList = exports.importsMapImageEntryList = void 0;
const baseEntryList_1 = require("./baseEntryList");
const imageEntry_1 = require("./imageEntry");
exports.importsMapImageEntryList = {
    BaseEntryList: baseEntryList_1.BaseEntryList,
    ImageEntry: imageEntry_1.ImageEntry,
};
/**
 * Represents a list of images which will be appended to the original resource document or image.
 */
class ImageEntryList extends baseEntryList_1.BaseEntryList {
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return super.getAttributeTypeMap().concat(ImageEntryList.attributeTypeMap);
    }
    constructor(init) {
        super(init);
        Object.assign(this, init);
    }
    collectFilesContent(_resultFilesContent) {
        super.collectFilesContent(_resultFilesContent);
        if (this.imageEntries) {
            for (let element of this.imageEntries) {
                element.collectFilesContent(_resultFilesContent);
            }
        }
    }
    validate() {
        super.validate();
        if (this.imageEntries === null || this.imageEntries === undefined) {
            throw new Error('Property ImageEntries in ImageEntryList is required.');
        }
        if (this.imageEntries !== null && this.imageEntries !== undefined) {
            for (let elementImageEntries of this.imageEntries) {
                elementImageEntries === null || elementImageEntries === void 0 ? void 0 : elementImageEntries.validate();
            }
        }
    }
}
exports.ImageEntryList = ImageEntryList;
/**
 * Attribute type map
 */
ImageEntryList.attributeTypeMap = [
    {
        name: "appendEachImageOnNewPage",
        baseName: "AppendEachImageOnNewPage",
        type: "boolean",
    },
    {
        name: "imageEntries",
        baseName: "ImageEntries",
        type: "Array<ImageEntry>",
    }
];
