// asposewordscloud/dist/model/jsonDataLoadOptions.js
"use strict";
/*
 * --------------------------------------------------------------------------------
 * <copyright company="Aspose" file="jsonDataLoadOptions.ts">
 *   Copyright (c) 2024 Aspose.Words for Cloud
 * </copyright>
 * <summary>
 *   Permission is hereby granted, free of charge, to any person obtaining a copy
 *  of this software and associated documentation files (the "Software"), to deal
 *  in the Software without restriction, including without limitation the rights
 *  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 *  copies of the Software, and to permit persons to whom the Software is
 *  furnished to do so, subject to the following conditions:
 *
 *  The above copyright notice and this permission notice shall be included in all
 *  copies or substantial portions of the Software.
 *
 *  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 *  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 *  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 *  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 *  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 *  SOFTWARE.
 * </summary>
 * --------------------------------------------------------------------------------
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.JsonDataLoadOptions = exports.importsMapJsonDataLoadOptions = void 0;
exports.importsMapJsonDataLoadOptions = {};
/**
 * Represents options for parsing JSON data.
 * An instance of this class can be passed into constructors of Aspose.Words.Reporting.JsonDataSource.
 */
class JsonDataLoadOptions {
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return JsonDataLoadOptions.attributeTypeMap;
    }
    constructor(init) {
        Object.assign(this, init);
    }
    collectFilesContent(_resultFilesContent) {
    }
    validate() {
        if (this.alwaysGenerateRootObject === null || this.alwaysGenerateRootObject === undefined) {
            throw new Error('Property AlwaysGenerateRootObject in JsonDataLoadOptions is required.');
        }
        if (this.simpleValueParseMode === null || this.simpleValueParseMode === undefined) {
            throw new Error('Property SimpleValueParseMode in JsonDataLoadOptions is required.');
        }
    }
}
exports.JsonDataLoadOptions = JsonDataLoadOptions;
/**
 * Attribute type map
 */
JsonDataLoadOptions.attributeTypeMap = [
    {
        name: "alwaysGenerateRootObject",
        baseName: "AlwaysGenerateRootObject",
        type: "boolean",
    },
    {
        name: "exactDateTimeParseFormats",
        baseName: "ExactDateTimeParseFormats",
        type: "Array<string>",
    },
    {
        name: "simpleValueParseMode",
        baseName: "SimpleValueParseMode",
        type: "JsonDataLoadOptions.SimpleValueParseModeEnum",
    }
];
/**
 * Enums for JsonDataLoadOptions
 */
// tslint:disable:quotemark
// tslint:disable-next-line:no-namespace
(function (JsonDataLoadOptions) {
    let SimpleValueParseModeEnum;
    (function (SimpleValueParseModeEnum) {
        SimpleValueParseModeEnum[SimpleValueParseModeEnum["Loose"] = 'Loose'] = "Loose";
        SimpleValueParseModeEnum[SimpleValueParseModeEnum["Strict"] = 'Strict'] = "Strict";
    })(SimpleValueParseModeEnum = JsonDataLoadOptions.SimpleValueParseModeEnum || (JsonDataLoadOptions.SimpleValueParseModeEnum = {}));
})(JsonDataLoadOptions = exports.JsonDataLoadOptions || (exports.JsonDataLoadOptions = {}));
// tslint:enable:quotemark
