// asposewordscloud/dist/model/listFormat.js
"use strict";
/*
 * --------------------------------------------------------------------------------
 * <copyright company="Aspose" file="listFormat.ts">
 *   Copyright (c) 2024 Aspose.Words for Cloud
 * </copyright>
 * <summary>
 *   Permission is hereby granted, free of charge, to any person obtaining a copy
 *  of this software and associated documentation files (the "Software"), to deal
 *  in the Software without restriction, including without limitation the rights
 *  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 *  copies of the Software, and to permit persons to whom the Software is
 *  furnished to do so, subject to the following conditions:
 *
 *  The above copyright notice and this permission notice shall be included in all
 *  copies or substantial portions of the Software.
 *
 *  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 *  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 *  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 *  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 *  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 *  SOFTWARE.
 * </summary>
 * --------------------------------------------------------------------------------
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListFormat = exports.importsMapListFormat = void 0;
const linkElement_1 = require("./linkElement");
exports.importsMapListFormat = {
    LinkElement: linkElement_1.LinkElement,
};
/**
 * DTO container with a paragraph list format element.
 */
class ListFormat extends linkElement_1.LinkElement {
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return super.getAttributeTypeMap().concat(ListFormat.attributeTypeMap);
    }
    constructor(init) {
        super(init);
        Object.assign(this, init);
    }
    collectFilesContent(_resultFilesContent) {
    }
    validate() {
        super.validate();
        if (this.listLevelNumber === null || this.listLevelNumber === undefined) {
            throw new Error('Property ListLevelNumber in ListFormat is required.');
        }
        if (this.isListItem === null || this.isListItem === undefined) {
            throw new Error('Property IsListItem in ListFormat is required.');
        }
    }
}
exports.ListFormat = ListFormat;
/**
 * Attribute type map
 */
ListFormat.attributeTypeMap = [
    {
        name: "listLevelNumber",
        baseName: "ListLevelNumber",
        type: "number",
    },
    {
        name: "listId",
        baseName: "ListId",
        type: "number",
    },
    {
        name: "isListItem",
        baseName: "IsListItem",
        type: "boolean",
    }
];
