// asposewordscloud/dist/model/loadWebDocumentData.js
"use strict";
/*
 * --------------------------------------------------------------------------------
 * <copyright company="Aspose" file="loadWebDocumentData.ts">
 *   Copyright (c) 2024 Aspose.Words for Cloud
 * </copyright>
 * <summary>
 *   Permission is hereby granted, free of charge, to any person obtaining a copy
 *  of this software and associated documentation files (the "Software"), to deal
 *  in the Software without restriction, including without limitation the rights
 *  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 *  copies of the Software, and to permit persons to whom the Software is
 *  furnished to do so, subject to the following conditions:
 *
 *  The above copyright notice and this permission notice shall be included in all
 *  copies or substantial portions of the Software.
 *
 *  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 *  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 *  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 *  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 *  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 *  SOFTWARE.
 * </summary>
 * --------------------------------------------------------------------------------
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LoadWebDocumentData = exports.importsMapLoadWebDocumentData = void 0;
const saveOptionsData_1 = require("./saveOptionsData");
exports.importsMapLoadWebDocumentData = {
    SaveOptionsData: saveOptionsData_1.SaveOptionsData,
};
/**
 * Contains data for load web document.
 */
class LoadWebDocumentData {
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return LoadWebDocumentData.attributeTypeMap;
    }
    constructor(init) {
        Object.assign(this, init);
    }
    collectFilesContent(_resultFilesContent) {
    }
    validate() {
        var _a;
        if (this.loadingDocumentUrl === null || this.loadingDocumentUrl === undefined) {
            throw new Error('Property LoadingDocumentUrl in LoadWebDocumentData is required.');
        }
        (_a = this.saveOptions) === null || _a === void 0 ? void 0 : _a.validate();
    }
}
exports.LoadWebDocumentData = LoadWebDocumentData;
/**
 * Attribute type map
 */
LoadWebDocumentData.attributeTypeMap = [
    {
        name: "saveOptions",
        baseName: "SaveOptions",
        type: "SaveOptionsData",
    },
    {
        name: "loadingDocumentUrl",
        baseName: "LoadingDocumentUrl",
        type: "string",
    }
];
