// asposewordscloud/dist/model/odtSaveOptionsData.js
"use strict";
/*
 * --------------------------------------------------------------------------------
 * <copyright company="Aspose" file="odtSaveOptionsData.ts">
 *   Copyright (c) 2024 Aspose.Words for Cloud
 * </copyright>
 * <summary>
 *   Permission is hereby granted, free of charge, to any person obtaining a copy
 *  of this software and associated documentation files (the "Software"), to deal
 *  in the Software without restriction, including without limitation the rights
 *  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 *  copies of the Software, and to permit persons to whom the Software is
 *  furnished to do so, subject to the following conditions:
 *
 *  The above copyright notice and this permission notice shall be included in all
 *  copies or substantial portions of the Software.
 *
 *  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 *  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 *  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 *  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 *  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 *  SOFTWARE.
 * </summary>
 * --------------------------------------------------------------------------------
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.OdtSaveOptionsData = exports.importsMapOdtSaveOptionsData = void 0;
const saveOptionsData_1 = require("./saveOptionsData");
exports.importsMapOdtSaveOptionsData = {
    SaveOptionsData: saveOptionsData_1.SaveOptionsData,
};
/**
 * Container class for odt/ott save options.
 */
class OdtSaveOptionsData extends saveOptionsData_1.SaveOptionsData {
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return super.getAttributeTypeMap().concat(OdtSaveOptionsData.attributeTypeMap);
    }
    constructor(init) {
        super(init);
        this.saveFormat = 'odt';
        Object.assign(this, init);
    }
    collectFilesContent(_resultFilesContent) {
    }
    validate() {
        super.validate();
    }
}
exports.OdtSaveOptionsData = OdtSaveOptionsData;
/**
 * Attribute type map
 */
OdtSaveOptionsData.attributeTypeMap = [
    {
        name: "isStrictSchema11",
        baseName: "IsStrictSchema11",
        type: "boolean",
    },
    {
        name: "measureUnit",
        baseName: "MeasureUnit",
        type: "OdtSaveOptionsData.MeasureUnitEnum",
    },
    {
        name: "password",
        baseName: "Password",
        type: "string",
    },
    {
        name: "prettyFormat",
        baseName: "PrettyFormat",
        type: "boolean",
    }
];
/**
 * Enums for OdtSaveOptionsData
 */
// tslint:disable:quotemark
// tslint:disable-next-line:no-namespace
(function (OdtSaveOptionsData) {
    let MeasureUnitEnum;
    (function (MeasureUnitEnum) {
        MeasureUnitEnum[MeasureUnitEnum["Centimeters"] = 'Centimeters'] = "Centimeters";
        MeasureUnitEnum[MeasureUnitEnum["Inches"] = 'Inches'] = "Inches";
    })(MeasureUnitEnum = OdtSaveOptionsData.MeasureUnitEnum || (OdtSaveOptionsData.MeasureUnitEnum = {}));
})(OdtSaveOptionsData = exports.OdtSaveOptionsData || (exports.OdtSaveOptionsData = {}));
// tslint:enable:quotemark
