// asposewordscloud/dist/model/ooxmlSaveOptionsData.js
"use strict";
/*
 * --------------------------------------------------------------------------------
 * <copyright company="Aspose" file="ooxmlSaveOptionsData.ts">
 *   Copyright (c) 2024 Aspose.Words for Cloud
 * </copyright>
 * <summary>
 *   Permission is hereby granted, free of charge, to any person obtaining a copy
 *  of this software and associated documentation files (the "Software"), to deal
 *  in the Software without restriction, including without limitation the rights
 *  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 *  copies of the Software, and to permit persons to whom the Software is
 *  furnished to do so, subject to the following conditions:
 *
 *  The above copyright notice and this permission notice shall be included in all
 *  copies or substantial portions of the Software.
 *
 *  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 *  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 *  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 *  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 *  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 *  SOFTWARE.
 * </summary>
 * --------------------------------------------------------------------------------
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.OoxmlSaveOptionsData = exports.importsMapOoxmlSaveOptionsData = void 0;
const saveOptionsData_1 = require("./saveOptionsData");
exports.importsMapOoxmlSaveOptionsData = {
    SaveOptionsData: saveOptionsData_1.SaveOptionsData,
};
/**
 * Container class for docx/docm/dotx/dotm/flatopc save options.
 */
class OoxmlSaveOptionsData extends saveOptionsData_1.SaveOptionsData {
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return super.getAttributeTypeMap().concat(OoxmlSaveOptionsData.attributeTypeMap);
    }
    constructor(init) {
        super(init);
        Object.assign(this, init);
    }
    collectFilesContent(_resultFilesContent) {
    }
    validate() {
        super.validate();
    }
}
exports.OoxmlSaveOptionsData = OoxmlSaveOptionsData;
/**
 * Attribute type map
 */
OoxmlSaveOptionsData.attributeTypeMap = [
    {
        name: "compliance",
        baseName: "Compliance",
        type: "OoxmlSaveOptionsData.ComplianceEnum",
    },
    {
        name: "compressionLevel",
        baseName: "CompressionLevel",
        type: "OoxmlSaveOptionsData.CompressionLevelEnum",
    },
    {
        name: "password",
        baseName: "Password",
        type: "string",
    },
    {
        name: "prettyFormat",
        baseName: "PrettyFormat",
        type: "boolean",
    }
];
/**
 * Enums for OoxmlSaveOptionsData
 */
// tslint:disable:quotemark
// tslint:disable-next-line:no-namespace
(function (OoxmlSaveOptionsData) {
    let ComplianceEnum;
    (function (ComplianceEnum) {
        ComplianceEnum[ComplianceEnum["Ecma376_2006"] = 'Ecma376_2006'] = "Ecma376_2006";
        ComplianceEnum[ComplianceEnum["Iso29500_2008_Transitional"] = 'Iso29500_2008_Transitional'] = "Iso29500_2008_Transitional";
        ComplianceEnum[ComplianceEnum["Iso29500_2008_Strict"] = 'Iso29500_2008_Strict'] = "Iso29500_2008_Strict";
    })(ComplianceEnum = OoxmlSaveOptionsData.ComplianceEnum || (OoxmlSaveOptionsData.ComplianceEnum = {}));
    let CompressionLevelEnum;
    (function (CompressionLevelEnum) {
        CompressionLevelEnum[CompressionLevelEnum["Normal"] = 'Normal'] = "Normal";
        CompressionLevelEnum[CompressionLevelEnum["Maximum"] = 'Maximum'] = "Maximum";
        CompressionLevelEnum[CompressionLevelEnum["Fast"] = 'Fast'] = "Fast";
        CompressionLevelEnum[CompressionLevelEnum["SuperFast"] = 'SuperFast'] = "SuperFast";
    })(CompressionLevelEnum = OoxmlSaveOptionsData.CompressionLevelEnum || (OoxmlSaveOptionsData.CompressionLevelEnum = {}));
})(OoxmlSaveOptionsData = exports.OoxmlSaveOptionsData || (exports.OoxmlSaveOptionsData = {}));
// tslint:enable:quotemark
