// asposewordscloud/dist/model/outlineOptionsData.js
"use strict";
/*
 * --------------------------------------------------------------------------------
 * <copyright company="Aspose" file="outlineOptionsData.ts">
 *   Copyright (c) 2024 Aspose.Words for Cloud
 * </copyright>
 * <summary>
 *   Permission is hereby granted, free of charge, to any person obtaining a copy
 *  of this software and associated documentation files (the "Software"), to deal
 *  in the Software without restriction, including without limitation the rights
 *  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 *  copies of the Software, and to permit persons to whom the Software is
 *  furnished to do so, subject to the following conditions:
 *
 *  The above copyright notice and this permission notice shall be included in all
 *  copies or substantial portions of the Software.
 *
 *  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 *  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 *  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 *  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 *  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 *  SOFTWARE.
 * </summary>
 * --------------------------------------------------------------------------------
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.OutlineOptionsData = exports.importsMapOutlineOptionsData = void 0;
const bookmarksOutlineLevelData_1 = require("./bookmarksOutlineLevelData");
exports.importsMapOutlineOptionsData = {
    BookmarksOutlineLevelData: bookmarksOutlineLevelData_1.BookmarksOutlineLevelData,
};
/**
 * Container class for outline options.
 */
class OutlineOptionsData {
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return OutlineOptionsData.attributeTypeMap;
    }
    constructor(init) {
        Object.assign(this, init);
    }
    collectFilesContent(_resultFilesContent) {
    }
    validate() {
        if (this.bookmarksOutlineLevels !== null && this.bookmarksOutlineLevels !== undefined) {
            for (let elementBookmarksOutlineLevels of this.bookmarksOutlineLevels) {
                elementBookmarksOutlineLevels === null || elementBookmarksOutlineLevels === void 0 ? void 0 : elementBookmarksOutlineLevels.validate();
            }
        }
    }
}
exports.OutlineOptionsData = OutlineOptionsData;
/**
 * Attribute type map
 */
OutlineOptionsData.attributeTypeMap = [
    {
        name: "createMissingOutlineLevels",
        baseName: "CreateMissingOutlineLevels",
        type: "boolean",
    },
    {
        name: "createOutlinesForHeadingsInTables",
        baseName: "CreateOutlinesForHeadingsInTables",
        type: "boolean",
    },
    {
        name: "defaultBookmarksOutlineLevel",
        baseName: "DefaultBookmarksOutlineLevel",
        type: "number",
    },
    {
        name: "expandedOutlineLevels",
        baseName: "ExpandedOutlineLevels",
        type: "number",
    },
    {
        name: "headingsOutlineLevels",
        baseName: "HeadingsOutlineLevels",
        type: "number",
    },
    {
        name: "bookmarksOutlineLevels",
        baseName: "BookmarksOutlineLevels",
        type: "Array<BookmarksOutlineLevelData>",
    }
];
