// asposewordscloud/dist/model/pageNumber.js
"use strict";
/*
 * --------------------------------------------------------------------------------
 * <copyright company="Aspose" file="pageNumber.ts">
 *   Copyright (c) 2024 Aspose.Words for Cloud
 * </copyright>
 * <summary>
 *   Permission is hereby granted, free of charge, to any person obtaining a copy
 *  of this software and associated documentation files (the "Software"), to deal
 *  in the Software without restriction, including without limitation the rights
 *  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 *  copies of the Software, and to permit persons to whom the Software is
 *  furnished to do so, subject to the following conditions:
 *
 *  The above copyright notice and this permission notice shall be included in all
 *  copies or substantial portions of the Software.
 *
 *  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 *  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 *  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 *  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 *  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 *  SOFTWARE.
 * </summary>
 * --------------------------------------------------------------------------------
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PageNumber = exports.importsMapPageNumber = void 0;
exports.importsMapPageNumber = {};
/**
 * Class is used for insert page number request building.
 */
class PageNumber {
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return PageNumber.attributeTypeMap;
    }
    constructor(init) {
        Object.assign(this, init);
    }
    collectFilesContent(_resultFilesContent) {
    }
    validate() {
        if (this.isTop === null || this.isTop === undefined) {
            throw new Error('Property IsTop in PageNumber is required.');
        }
        if (this.setPageNumberOnFirstPage === null || this.setPageNumberOnFirstPage === undefined) {
            throw new Error('Property SetPageNumberOnFirstPage in PageNumber is required.');
        }
    }
}
exports.PageNumber = PageNumber;
/**
 * Attribute type map
 */
PageNumber.attributeTypeMap = [
    {
        name: "alignment",
        baseName: "Alignment",
        type: "string",
    },
    {
        name: "format",
        baseName: "Format",
        type: "string",
    },
    {
        name: "isTop",
        baseName: "IsTop",
        type: "boolean",
    },
    {
        name: "pageStartingNumber",
        baseName: "PageStartingNumber",
        type: "number",
    },
    {
        name: "setPageNumberOnFirstPage",
        baseName: "SetPageNumberOnFirstPage",
        type: "boolean",
    }
];
