// asposewordscloud/dist/model/pdfPermissions.js
"use strict";
/*
 * --------------------------------------------------------------------------------
 * <copyright company="Aspose" file="pdfPermissions.ts">
 *   Copyright (c) 2024 Aspose.Words for Cloud
 * </copyright>
 * <summary>
 *   Permission is hereby granted, free of charge, to any person obtaining a copy
 *  of this software and associated documentation files (the "Software"), to deal
 *  in the Software without restriction, including without limitation the rights
 *  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 *  copies of the Software, and to permit persons to whom the Software is
 *  furnished to do so, subject to the following conditions:
 *
 *  The above copyright notice and this permission notice shall be included in all
 *  copies or substantial portions of the Software.
 *
 *  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 *  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 *  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 *  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 *  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 *  SOFTWARE.
 * </summary>
 * --------------------------------------------------------------------------------
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PdfPermissions = void 0;
var PdfPermissions;
(function (PdfPermissions) {
    PdfPermissions[PdfPermissions["DisallowAll"] = 'DisallowAll'] = "DisallowAll";
    PdfPermissions[PdfPermissions["Printing"] = 'Printing'] = "Printing";
    PdfPermissions[PdfPermissions["ModifyContents"] = 'ModifyContents'] = "ModifyContents";
    PdfPermissions[PdfPermissions["ContentCopy"] = 'ContentCopy'] = "ContentCopy";
    PdfPermissions[PdfPermissions["ModifyAnnotations"] = 'ModifyAnnotations'] = "ModifyAnnotations";
    PdfPermissions[PdfPermissions["FillIn"] = 'FillIn'] = "FillIn";
    PdfPermissions[PdfPermissions["ContentCopyForAccessibility"] = 'ContentCopyForAccessibility'] = "ContentCopyForAccessibility";
    PdfPermissions[PdfPermissions["DocumentAssembly"] = 'DocumentAssembly'] = "DocumentAssembly";
    PdfPermissions[PdfPermissions["HighResolutionPrinting"] = 'HighResolutionPrinting'] = "HighResolutionPrinting";
    PdfPermissions[PdfPermissions["AllowAll"] = 'AllowAll'] = "AllowAll";
})(PdfPermissions = exports.PdfPermissions || (exports.PdfPermissions = {}));
