// asposewordscloud/dist/model/protectionData.js
"use strict";
/*
 * --------------------------------------------------------------------------------
 * <copyright company="Aspose" file="protectionData.ts">
 *   Copyright (c) 2024 Aspose.Words for Cloud
 * </copyright>
 * <summary>
 *   Permission is hereby granted, free of charge, to any person obtaining a copy
 *  of this software and associated documentation files (the "Software"), to deal
 *  in the Software without restriction, including without limitation the rights
 *  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 *  copies of the Software, and to permit persons to whom the Software is
 *  furnished to do so, subject to the following conditions:
 *
 *  The above copyright notice and this permission notice shall be included in all
 *  copies or substantial portions of the Software.
 *
 *  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 *  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 *  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 *  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 *  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 *  SOFTWARE.
 * </summary>
 * --------------------------------------------------------------------------------
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProtectionData = exports.importsMapProtectionData = void 0;
exports.importsMapProtectionData = {};
/**
 * Container for the data about protection of the document.
 */
class ProtectionData {
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return ProtectionData.attributeTypeMap;
    }
    constructor(init) {
        Object.assign(this, init);
    }
    collectFilesContent(_resultFilesContent) {
    }
    validate() {
        if (this.protectionType === null || this.protectionType === undefined) {
            throw new Error('Property ProtectionType in ProtectionData is required.');
        }
    }
}
exports.ProtectionData = ProtectionData;
/**
 * Attribute type map
 */
ProtectionData.attributeTypeMap = [
    {
        name: "protectionType",
        baseName: "ProtectionType",
        type: "ProtectionData.ProtectionTypeEnum",
    }
];
/**
 * Enums for ProtectionData
 */
// tslint:disable:quotemark
// tslint:disable-next-line:no-namespace
(function (ProtectionData) {
    let ProtectionTypeEnum;
    (function (ProtectionTypeEnum) {
        ProtectionTypeEnum[ProtectionTypeEnum["AllowOnlyRevisions"] = 'AllowOnlyRevisions'] = "AllowOnlyRevisions";
        ProtectionTypeEnum[ProtectionTypeEnum["AllowOnlyComments"] = 'AllowOnlyComments'] = "AllowOnlyComments";
        ProtectionTypeEnum[ProtectionTypeEnum["AllowOnlyFormFields"] = 'AllowOnlyFormFields'] = "AllowOnlyFormFields";
        ProtectionTypeEnum[ProtectionTypeEnum["ReadOnly"] = 'ReadOnly'] = "ReadOnly";
        ProtectionTypeEnum[ProtectionTypeEnum["NoProtection"] = 'NoProtection'] = "NoProtection";
    })(ProtectionTypeEnum = ProtectionData.ProtectionTypeEnum || (ProtectionData.ProtectionTypeEnum = {}));
})(ProtectionData = exports.ProtectionData || (exports.ProtectionData = {}));
// tslint:enable:quotemark
