// asposewordscloud/dist/model/protectionRequest.js
"use strict";
/*
 * --------------------------------------------------------------------------------
 * <copyright company="Aspose" file="protectionRequest.ts">
 *   Copyright (c) 2024 Aspose.Words for Cloud
 * </copyright>
 * <summary>
 *   Permission is hereby granted, free of charge, to any person obtaining a copy
 *  of this software and associated documentation files (the "Software"), to deal
 *  in the Software without restriction, including without limitation the rights
 *  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 *  copies of the Software, and to permit persons to whom the Software is
 *  furnished to do so, subject to the following conditions:
 *
 *  The above copyright notice and this permission notice shall be included in all
 *  copies or substantial portions of the Software.
 *
 *  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 *  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 *  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 *  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 *  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 *  SOFTWARE.
 * </summary>
 * --------------------------------------------------------------------------------
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProtectionRequest = exports.importsMapProtectionRequest = void 0;
const protectionRequestBase_1 = require("./protectionRequestBase");
exports.importsMapProtectionRequest = {
    ProtectionRequestBase: protectionRequestBase_1.ProtectionRequestBase,
};
/**
 * Request on changing of protection.
 * @deprecated ProtectionRequest is deprecated and remains for backwards compatibility only.
 */
class ProtectionRequest extends protectionRequestBase_1.ProtectionRequestBase {
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return super.getAttributeTypeMap().concat(ProtectionRequest.attributeTypeMap);
    }
    constructor(init) {
        super(init);
        Object.assign(this, init);
    }
    collectFilesContent(_resultFilesContent) {
    }
    validate() {
        super.validate();
        if (this.password === null || this.password === undefined) {
            throw new Error('Property Password in ProtectionRequest is required.');
        }
    }
}
exports.ProtectionRequest = ProtectionRequest;
/**
 * Attribute type map
 */
ProtectionRequest.attributeTypeMap = [
    {
        name: "newPassword",
        baseName: "NewPassword",
        type: "string",
    },
    {
        name: "password",
        baseName: "Password",
        type: "string",
    },
    {
        name: "protectionType",
        baseName: "ProtectionType",
        type: "string",
    }
];
