// asposewordscloud/dist/model/rangeTextResponse.js
"use strict";
/*
 * --------------------------------------------------------------------------------
 * <copyright company="Aspose" file="rangeTextResponse.ts">
 *   Copyright (c) 2024 Aspose.Words for Cloud
 * </copyright>
 * <summary>
 *   Permission is hereby granted, free of charge, to any person obtaining a copy
 *  of this software and associated documentation files (the "Software"), to deal
 *  in the Software without restriction, including without limitation the rights
 *  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 *  copies of the Software, and to permit persons to whom the Software is
 *  furnished to do so, subject to the following conditions:
 *
 *  The above copyright notice and this permission notice shall be included in all
 *  copies or substantial portions of the Software.
 *
 *  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 *  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 *  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 *  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 *  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 *  SOFTWARE.
 * </summary>
 * --------------------------------------------------------------------------------
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.RangeTextResponse = exports.importsMapRangeTextResponse = void 0;
const wordsResponse_1 = require("./wordsResponse");
exports.importsMapRangeTextResponse = {
    WordsResponse: wordsResponse_1.WordsResponse,
};
/**
 * The REST response with a range's text.
 * This response is returned by the Service when handling "GET https://api.aspose.cloud/v4.0/words/Test.doc/range/{0}/{1}/" REST API requests.
 */
class RangeTextResponse extends wordsResponse_1.WordsResponse {
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return super.getAttributeTypeMap().concat(RangeTextResponse.attributeTypeMap);
    }
    constructor(init) {
        super(init);
        Object.assign(this, init);
    }
    collectFilesContent(_resultFilesContent) {
    }
    validate() {
        super.validate();
    }
}
exports.RangeTextResponse = RangeTextResponse;
/**
 * Attribute type map
 */
RangeTextResponse.attributeTypeMap = [
    {
        name: "text",
        baseName: "Text",
        type: "string",
    }
];
