// asposewordscloud/dist/model/replaceRange.js
"use strict";
/*
 * --------------------------------------------------------------------------------
 * <copyright company="Aspose" file="replaceRange.ts">
 *   Copyright (c) 2024 Aspose.Words for Cloud
 * </copyright>
 * <summary>
 *   Permission is hereby granted, free of charge, to any person obtaining a copy
 *  of this software and associated documentation files (the "Software"), to deal
 *  in the Software without restriction, including without limitation the rights
 *  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 *  copies of the Software, and to permit persons to whom the Software is
 *  furnished to do so, subject to the following conditions:
 *
 *  The above copyright notice and this permission notice shall be included in all
 *  copies or substantial portions of the Software.
 *
 *  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 *  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 *  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 *  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 *  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 *  SOFTWARE.
 * </summary>
 * --------------------------------------------------------------------------------
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ReplaceRange = exports.importsMapReplaceRange = void 0;
exports.importsMapReplaceRange = {};
/**
 * DTO container with a range element.
 */
class ReplaceRange {
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return ReplaceRange.attributeTypeMap;
    }
    constructor(init) {
        Object.assign(this, init);
    }
    collectFilesContent(_resultFilesContent) {
    }
    validate() {
        if (this.text === null || this.text === undefined) {
            throw new Error('Property Text in ReplaceRange is required.');
        }
    }
}
exports.ReplaceRange = ReplaceRange;
/**
 * Attribute type map
 */
ReplaceRange.attributeTypeMap = [
    {
        name: "text",
        baseName: "Text",
        type: "string",
    },
    {
        name: "textType",
        baseName: "TextType",
        type: "ReplaceRange.TextTypeEnum",
    }
];
/**
 * Enums for ReplaceRange
 */
// tslint:disable:quotemark
// tslint:disable-next-line:no-namespace
(function (ReplaceRange) {
    let TextTypeEnum;
    (function (TextTypeEnum) {
        TextTypeEnum[TextTypeEnum["Text"] = 'Text'] = "Text";
        TextTypeEnum[TextTypeEnum["Html"] = 'Html'] = "Html";
    })(TextTypeEnum = ReplaceRange.TextTypeEnum || (ReplaceRange.TextTypeEnum = {}));
})(ReplaceRange = exports.ReplaceRange || (exports.ReplaceRange = {}));
// tslint:enable:quotemark
