// asposewordscloud/dist/model/replaceTextResponse.js
"use strict";
/*
 * --------------------------------------------------------------------------------
 * <copyright company="Aspose" file="replaceTextResponse.ts">
 *   Copyright (c) 2024 Aspose.Words for Cloud
 * </copyright>
 * <summary>
 *   Permission is hereby granted, free of charge, to any person obtaining a copy
 *  of this software and associated documentation files (the "Software"), to deal
 *  in the Software without restriction, including without limitation the rights
 *  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 *  copies of the Software, and to permit persons to whom the Software is
 *  furnished to do so, subject to the following conditions:
 *
 *  The above copyright notice and this permission notice shall be included in all
 *  copies or substantial portions of the Software.
 *
 *  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 *  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 *  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 *  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 *  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 *  SOFTWARE.
 * </summary>
 * --------------------------------------------------------------------------------
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ReplaceTextResponse = exports.importsMapReplaceTextResponse = void 0;
const fileLink_1 = require("./fileLink");
const wordsResponse_1 = require("./wordsResponse");
exports.importsMapReplaceTextResponse = {
    FileLink: fileLink_1.FileLink,
    WordsResponse: wordsResponse_1.WordsResponse,
};
/**
 * The REST response with a number of occurrences of the captured text in the document.
 */
class ReplaceTextResponse extends wordsResponse_1.WordsResponse {
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return super.getAttributeTypeMap().concat(ReplaceTextResponse.attributeTypeMap);
    }
    constructor(init) {
        super(init);
        Object.assign(this, init);
    }
    collectFilesContent(_resultFilesContent) {
    }
    validate() {
        var _a;
        super.validate();
        if (this.matches === null || this.matches === undefined) {
            throw new Error('Property Matches in ReplaceTextResponse is required.');
        }
        (_a = this.documentLink) === null || _a === void 0 ? void 0 : _a.validate();
    }
}
exports.ReplaceTextResponse = ReplaceTextResponse;
/**
 * Attribute type map
 */
ReplaceTextResponse.attributeTypeMap = [
    {
        name: "documentLink",
        baseName: "DocumentLink",
        type: "FileLink",
    },
    {
        name: "matches",
        baseName: "Matches",
        type: "number",
    }
];
