// asposewordscloud/dist/model/searchResult.js
"use strict";
/*
 * --------------------------------------------------------------------------------
 * <copyright company="Aspose" file="searchResult.ts">
 *   Copyright (c) 2024 Aspose.Words for Cloud
 * </copyright>
 * <summary>
 *   Permission is hereby granted, free of charge, to any person obtaining a copy
 *  of this software and associated documentation files (the "Software"), to deal
 *  in the Software without restriction, including without limitation the rights
 *  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 *  copies of the Software, and to permit persons to whom the Software is
 *  furnished to do so, subject to the following conditions:
 *
 *  The above copyright notice and this permission notice shall be included in all
 *  copies or substantial portions of the Software.
 *
 *  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 *  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 *  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 *  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 *  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 *  SOFTWARE.
 * </summary>
 * --------------------------------------------------------------------------------
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SearchResult = exports.importsMapSearchResult = void 0;
const documentPosition_1 = require("./documentPosition");
exports.importsMapSearchResult = {
    DocumentPosition: documentPosition_1.DocumentPosition,
};
/**
 * Result of search operation.
 */
class SearchResult {
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return SearchResult.attributeTypeMap;
    }
    constructor(init) {
        Object.assign(this, init);
    }
    collectFilesContent(_resultFilesContent) {
    }
    validate() {
        var _a, _b;
        (_a = this.rangeStart) === null || _a === void 0 ? void 0 : _a.validate();
        (_b = this.rangeEnd) === null || _b === void 0 ? void 0 : _b.validate();
    }
}
exports.SearchResult = SearchResult;
/**
 * Attribute type map
 */
SearchResult.attributeTypeMap = [
    {
        name: "rangeStart",
        baseName: "RangeStart",
        type: "DocumentPosition",
    },
    {
        name: "rangeEnd",
        baseName: "RangeEnd",
        type: "DocumentPosition",
    }
];
