// asposewordscloud/dist/model/storyChildNodes.js
"use strict";
/*
 * --------------------------------------------------------------------------------
 * <copyright company="Aspose" file="storyChildNodes.ts">
 *   Copyright (c) 2024 Aspose.Words for Cloud
 * </copyright>
 * <summary>
 *   Permission is hereby granted, free of charge, to any person obtaining a copy
 *  of this software and associated documentation files (the "Software"), to deal
 *  in the Software without restriction, including without limitation the rights
 *  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 *  copies of the Software, and to permit persons to whom the Software is
 *  furnished to do so, subject to the following conditions:
 *
 *  The above copyright notice and this permission notice shall be included in all
 *  copies or substantial portions of the Software.
 *
 *  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 *  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 *  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 *  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 *  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 *  SOFTWARE.
 * </summary>
 * --------------------------------------------------------------------------------
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.StoryChildNodes = exports.importsMapStoryChildNodes = void 0;
const nodeLink_1 = require("./nodeLink");
exports.importsMapStoryChildNodes = {
    NodeLink: nodeLink_1.NodeLink,
};
/**
 * Child nodes of Story or InlineStory.
 */
class StoryChildNodes {
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return StoryChildNodes.attributeTypeMap;
    }
    constructor(init) {
        Object.assign(this, init);
    }
    collectFilesContent(_resultFilesContent) {
    }
    validate() {
        if (this.childNodes !== null && this.childNodes !== undefined) {
            for (let elementChildNodes of this.childNodes) {
                elementChildNodes === null || elementChildNodes === void 0 ? void 0 : elementChildNodes.validate();
            }
        }
    }
}
exports.StoryChildNodes = StoryChildNodes;
/**
 * Attribute type map
 */
StoryChildNodes.attributeTypeMap = [
    {
        name: "childNodes",
        baseName: "ChildNodes",
        type: "Array<NodeLink>",
    }
];
