// asposewordscloud/dist/model/structuredDocumentTag.js
"use strict";
/*
 * --------------------------------------------------------------------------------
 * <copyright company="Aspose" file="structuredDocumentTag.ts">
 *   Copyright (c) 2024 Aspose.Words for Cloud
 * </copyright>
 * <summary>
 *   Permission is hereby granted, free of charge, to any person obtaining a copy
 *  of this software and associated documentation files (the "Software"), to deal
 *  in the Software without restriction, including without limitation the rights
 *  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 *  copies of the Software, and to permit persons to whom the Software is
 *  furnished to do so, subject to the following conditions:
 *
 *  The above copyright notice and this permission notice shall be included in all
 *  copies or substantial portions of the Software.
 *
 *  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 *  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 *  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 *  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 *  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 *  SOFTWARE.
 * </summary>
 * --------------------------------------------------------------------------------
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.StructuredDocumentTag = exports.importsMapStructuredDocumentTag = void 0;
const structuredDocumentTagBase_1 = require("./structuredDocumentTagBase");
exports.importsMapStructuredDocumentTag = {
    StructuredDocumentTagBase: structuredDocumentTagBase_1.StructuredDocumentTagBase,
};
/**
 * DTO container with a StructuredDocumentTag.
 */
class StructuredDocumentTag extends structuredDocumentTagBase_1.StructuredDocumentTagBase {
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return super.getAttributeTypeMap().concat(StructuredDocumentTag.attributeTypeMap);
    }
    constructor(init) {
        super(init);
        Object.assign(this, init);
    }
    collectFilesContent(_resultFilesContent) {
    }
    validate() {
        super.validate();
    }
}
exports.StructuredDocumentTag = StructuredDocumentTag;
/**
 * Attribute type map
 */
StructuredDocumentTag.attributeTypeMap = [
    {
        name: "level",
        baseName: "Level",
        type: "StructuredDocumentTag.LevelEnum",
    },
    {
        name: "sdtType",
        baseName: "SdtType",
        type: "StructuredDocumentTag.SdtTypeEnum",
    }
];
/**
 * Enums for StructuredDocumentTag
 */
// tslint:disable:quotemark
// tslint:disable-next-line:no-namespace
(function (StructuredDocumentTag) {
    let LevelEnum;
    (function (LevelEnum) {
        LevelEnum[LevelEnum["Unknown"] = 'Unknown'] = "Unknown";
        LevelEnum[LevelEnum["Inline"] = 'Inline'] = "Inline";
        LevelEnum[LevelEnum["Block"] = 'Block'] = "Block";
        LevelEnum[LevelEnum["Row"] = 'Row'] = "Row";
        LevelEnum[LevelEnum["Cell"] = 'Cell'] = "Cell";
    })(LevelEnum = StructuredDocumentTag.LevelEnum || (StructuredDocumentTag.LevelEnum = {}));
    let SdtTypeEnum;
    (function (SdtTypeEnum) {
        SdtTypeEnum[SdtTypeEnum["None"] = 'None'] = "None";
        SdtTypeEnum[SdtTypeEnum["Bibliography"] = 'Bibliography'] = "Bibliography";
        SdtTypeEnum[SdtTypeEnum["Citation"] = 'Citation'] = "Citation";
        SdtTypeEnum[SdtTypeEnum["Equation"] = 'Equation'] = "Equation";
        SdtTypeEnum[SdtTypeEnum["DropDownList"] = 'DropDownList'] = "DropDownList";
        SdtTypeEnum[SdtTypeEnum["ComboBox"] = 'ComboBox'] = "ComboBox";
        SdtTypeEnum[SdtTypeEnum["Date"] = 'Date'] = "Date";
        SdtTypeEnum[SdtTypeEnum["BuildingBlockGallery"] = 'BuildingBlockGallery'] = "BuildingBlockGallery";
        SdtTypeEnum[SdtTypeEnum["DocPartObj"] = 'DocPartObj'] = "DocPartObj";
        SdtTypeEnum[SdtTypeEnum["Group"] = 'Group'] = "Group";
        SdtTypeEnum[SdtTypeEnum["Picture"] = 'Picture'] = "Picture";
        SdtTypeEnum[SdtTypeEnum["RichText"] = 'RichText'] = "RichText";
        SdtTypeEnum[SdtTypeEnum["PlainText"] = 'PlainText'] = "PlainText";
        SdtTypeEnum[SdtTypeEnum["Checkbox"] = 'Checkbox'] = "Checkbox";
        SdtTypeEnum[SdtTypeEnum["RepeatingSection"] = 'RepeatingSection'] = "RepeatingSection";
        SdtTypeEnum[SdtTypeEnum["RepeatingSectionItem"] = 'RepeatingSectionItem'] = "RepeatingSectionItem";
        SdtTypeEnum[SdtTypeEnum["EntityPicker"] = 'EntityPicker'] = "EntityPicker";
    })(SdtTypeEnum = StructuredDocumentTag.SdtTypeEnum || (StructuredDocumentTag.SdtTypeEnum = {}));
})(StructuredDocumentTag = exports.StructuredDocumentTag || (exports.StructuredDocumentTag = {}));
// tslint:enable:quotemark
