// asposewordscloud/dist/model/styleInsert.js
"use strict";
/*
 * --------------------------------------------------------------------------------
 * <copyright company="Aspose" file="styleInsert.ts">
 *   Copyright (c) 2024 Aspose.Words for Cloud
 * </copyright>
 * <summary>
 *   Permission is hereby granted, free of charge, to any person obtaining a copy
 *  of this software and associated documentation files (the "Software"), to deal
 *  in the Software without restriction, including without limitation the rights
 *  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 *  copies of the Software, and to permit persons to whom the Software is
 *  furnished to do so, subject to the following conditions:
 *
 *  The above copyright notice and this permission notice shall be included in all
 *  copies or substantial portions of the Software.
 *
 *  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 *  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 *  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 *  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 *  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 *  SOFTWARE.
 * </summary>
 * --------------------------------------------------------------------------------
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.StyleInsert = exports.importsMapStyleInsert = void 0;
exports.importsMapStyleInsert = {};
/**
 * Represents a single document style to insert.
 */
class StyleInsert {
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return StyleInsert.attributeTypeMap;
    }
    constructor(init) {
        Object.assign(this, init);
    }
    collectFilesContent(_resultFilesContent) {
    }
    validate() {
        if (this.styleName === null || this.styleName === undefined) {
            throw new Error('Property StyleName in StyleInsert is required.');
        }
        if (this.styleType === null || this.styleType === undefined) {
            throw new Error('Property StyleType in StyleInsert is required.');
        }
    }
}
exports.StyleInsert = StyleInsert;
/**
 * Attribute type map
 */
StyleInsert.attributeTypeMap = [
    {
        name: "styleName",
        baseName: "StyleName",
        type: "string",
    },
    {
        name: "styleType",
        baseName: "StyleType",
        type: "StyleInsert.StyleTypeEnum",
    }
];
/**
 * Enums for StyleInsert
 */
// tslint:disable:quotemark
// tslint:disable-next-line:no-namespace
(function (StyleInsert) {
    let StyleTypeEnum;
    (function (StyleTypeEnum) {
        StyleTypeEnum[StyleTypeEnum["Paragraph"] = 'Paragraph'] = "Paragraph";
        StyleTypeEnum[StyleTypeEnum["Character"] = 'Character'] = "Character";
        StyleTypeEnum[StyleTypeEnum["Table"] = 'Table'] = "Table";
        StyleTypeEnum[StyleTypeEnum["List"] = 'List'] = "List";
    })(StyleTypeEnum = StyleInsert.StyleTypeEnum || (StyleInsert.StyleTypeEnum = {}));
})(StyleInsert = exports.StyleInsert || (exports.StyleInsert = {}));
// tslint:enable:quotemark
