// asposewordscloud/dist/model/table.js
"use strict";
/*
 * --------------------------------------------------------------------------------
 * <copyright company="Aspose" file="table.ts">
 *   Copyright (c) 2024 Aspose.Words for Cloud
 * </copyright>
 * <summary>
 *   Permission is hereby granted, free of charge, to any person obtaining a copy
 *  of this software and associated documentation files (the "Software"), to deal
 *  in the Software without restriction, including without limitation the rights
 *  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 *  copies of the Software, and to permit persons to whom the Software is
 *  furnished to do so, subject to the following conditions:
 *
 *  The above copyright notice and this permission notice shall be included in all
 *  copies or substantial portions of the Software.
 *
 *  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 *  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 *  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 *  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 *  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 *  SOFTWARE.
 * </summary>
 * --------------------------------------------------------------------------------
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Table = exports.importsMapTable = void 0;
const nodeLink_1 = require("./nodeLink");
const tableProperties_1 = require("./tableProperties");
const tableRow_1 = require("./tableRow");
exports.importsMapTable = {
    NodeLink: nodeLink_1.NodeLink,
    TableProperties: tableProperties_1.TableProperties,
    TableRow: tableRow_1.TableRow,
};
/**
 * DTO container with a table element.
 */
class Table extends nodeLink_1.NodeLink {
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return super.getAttributeTypeMap().concat(Table.attributeTypeMap);
    }
    constructor(init) {
        super(init);
        Object.assign(this, init);
    }
    collectFilesContent(_resultFilesContent) {
    }
    validate() {
        var _a;
        super.validate();
        if (this.tableRowList !== null && this.tableRowList !== undefined) {
            for (let elementTableRowList of this.tableRowList) {
                elementTableRowList === null || elementTableRowList === void 0 ? void 0 : elementTableRowList.validate();
            }
        }
        (_a = this.tableProperties) === null || _a === void 0 ? void 0 : _a.validate();
    }
}
exports.Table = Table;
/**
 * Attribute type map
 */
Table.attributeTypeMap = [
    {
        name: "tableRowList",
        baseName: "TableRowList",
        type: "Array<TableRow>",
    },
    {
        name: "tableProperties",
        baseName: "TableProperties",
        type: "TableProperties",
    }
];
