// asposewordscloud/dist/model/tableInsert.js
"use strict";
/*
 * --------------------------------------------------------------------------------
 * <copyright company="Aspose" file="tableInsert.ts">
 *   Copyright (c) 2024 Aspose.Words for Cloud
 * </copyright>
 * <summary>
 *   Permission is hereby granted, free of charge, to any person obtaining a copy
 *  of this software and associated documentation files (the "Software"), to deal
 *  in the Software without restriction, including without limitation the rights
 *  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 *  copies of the Software, and to permit persons to whom the Software is
 *  furnished to do so, subject to the following conditions:
 *
 *  The above copyright notice and this permission notice shall be included in all
 *  copies or substantial portions of the Software.
 *
 *  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 *  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 *  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 *  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 *  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 *  SOFTWARE.
 * </summary>
 * --------------------------------------------------------------------------------
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TableInsert = exports.importsMapTableInsert = void 0;
const position_1 = require("./position");
exports.importsMapTableInsert = {
    Position: position_1.Position,
};
/**
 * DTO container with a table element.
 */
class TableInsert {
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return TableInsert.attributeTypeMap;
    }
    constructor(init) {
        Object.assign(this, init);
    }
    collectFilesContent(_resultFilesContent) {
    }
    validate() {
        var _a;
        if (this.columnsCount === null || this.columnsCount === undefined) {
            throw new Error('Property ColumnsCount in TableInsert is required.');
        }
        if (this.rowsCount === null || this.rowsCount === undefined) {
            throw new Error('Property RowsCount in TableInsert is required.');
        }
        (_a = this.position) === null || _a === void 0 ? void 0 : _a.validate();
    }
}
exports.TableInsert = TableInsert;
/**
 * Attribute type map
 */
TableInsert.attributeTypeMap = [
    {
        name: "columnsCount",
        baseName: "ColumnsCount",
        type: "number",
    },
    {
        name: "position",
        baseName: "Position",
        type: "Position",
    },
    {
        name: "rowsCount",
        baseName: "RowsCount",
        type: "number",
    }
];
