// asposewordscloud/dist/model/tableRowFormat.js
"use strict";
/*
 * --------------------------------------------------------------------------------
 * <copyright company="Aspose" file="tableRowFormat.ts">
 *   Copyright (c) 2024 Aspose.Words for Cloud
 * </copyright>
 * <summary>
 *   Permission is hereby granted, free of charge, to any person obtaining a copy
 *  of this software and associated documentation files (the "Software"), to deal
 *  in the Software without restriction, including without limitation the rights
 *  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 *  copies of the Software, and to permit persons to whom the Software is
 *  furnished to do so, subject to the following conditions:
 *
 *  The above copyright notice and this permission notice shall be included in all
 *  copies or substantial portions of the Software.
 *
 *  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 *  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 *  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 *  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 *  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 *  SOFTWARE.
 * </summary>
 * --------------------------------------------------------------------------------
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TableRowFormat = exports.importsMapTableRowFormat = void 0;
const linkElement_1 = require("./linkElement");
exports.importsMapTableRowFormat = {
    LinkElement: linkElement_1.LinkElement,
};
/**
 * DTO container with formatting for a table row.
 */
class TableRowFormat extends linkElement_1.LinkElement {
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return super.getAttributeTypeMap().concat(TableRowFormat.attributeTypeMap);
    }
    constructor(init) {
        super(init);
        Object.assign(this, init);
    }
    collectFilesContent(_resultFilesContent) {
    }
    validate() {
        super.validate();
    }
}
exports.TableRowFormat = TableRowFormat;
/**
 * Attribute type map
 */
TableRowFormat.attributeTypeMap = [
    {
        name: "height",
        baseName: "Height",
        type: "number",
    },
    {
        name: "heightRule",
        baseName: "HeightRule",
        type: "TableRowFormat.HeightRuleEnum",
    },
    {
        name: "allowBreakAcrossPages",
        baseName: "AllowBreakAcrossPages",
        type: "boolean",
    },
    {
        name: "headingFormat",
        baseName: "HeadingFormat",
        type: "boolean",
    }
];
/**
 * Enums for TableRowFormat
 */
// tslint:disable:quotemark
// tslint:disable-next-line:no-namespace
(function (TableRowFormat) {
    let HeightRuleEnum;
    (function (HeightRuleEnum) {
        HeightRuleEnum[HeightRuleEnum["AtLeast"] = 'AtLeast'] = "AtLeast";
        HeightRuleEnum[HeightRuleEnum["Exactly"] = 'Exactly'] = "Exactly";
        HeightRuleEnum[HeightRuleEnum["Auto"] = 'Auto'] = "Auto";
    })(HeightRuleEnum = TableRowFormat.HeightRuleEnum || (TableRowFormat.HeightRuleEnum = {}));
})(TableRowFormat = exports.TableRowFormat || (exports.TableRowFormat = {}));
// tslint:enable:quotemark
