// asposewordscloud/dist/model/textSaveOptionsData.js
"use strict";
/*
 * --------------------------------------------------------------------------------
 * <copyright company="Aspose" file="textSaveOptionsData.ts">
 *   Copyright (c) 2024 Aspose.Words for Cloud
 * </copyright>
 * <summary>
 *   Permission is hereby granted, free of charge, to any person obtaining a copy
 *  of this software and associated documentation files (the "Software"), to deal
 *  in the Software without restriction, including without limitation the rights
 *  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 *  copies of the Software, and to permit persons to whom the Software is
 *  furnished to do so, subject to the following conditions:
 *
 *  The above copyright notice and this permission notice shall be included in all
 *  copies or substantial portions of the Software.
 *
 *  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 *  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 *  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 *  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 *  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 *  SOFTWARE.
 * </summary>
 * --------------------------------------------------------------------------------
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TextSaveOptionsData = exports.importsMapTextSaveOptionsData = void 0;
const txtSaveOptionsBaseData_1 = require("./txtSaveOptionsBaseData");
exports.importsMapTextSaveOptionsData = {
    TxtSaveOptionsBaseData: txtSaveOptionsBaseData_1.TxtSaveOptionsBaseData,
};
/**
 * Container class for text save options.
 */
class TextSaveOptionsData extends txtSaveOptionsBaseData_1.TxtSaveOptionsBaseData {
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return super.getAttributeTypeMap().concat(TextSaveOptionsData.attributeTypeMap);
    }
    constructor(init) {
        super(init);
        this.saveFormat = 'txt';
        Object.assign(this, init);
    }
    collectFilesContent(_resultFilesContent) {
    }
    validate() {
        super.validate();
        if (this.maxCharactersPerLine === null || this.maxCharactersPerLine === undefined) {
            throw new Error('Property MaxCharactersPerLine in TextSaveOptionsData is required.');
        }
    }
}
exports.TextSaveOptionsData = TextSaveOptionsData;
/**
 * Attribute type map
 */
TextSaveOptionsData.attributeTypeMap = [
    {
        name: "addBidiMarks",
        baseName: "AddBidiMarks",
        type: "boolean",
    },
    {
        name: "maxCharactersPerLine",
        baseName: "MaxCharactersPerLine",
        type: "number",
    },
    {
        name: "preserveTableLayout",
        baseName: "PreserveTableLayout",
        type: "boolean",
    },
    {
        name: "simplifyListLabels",
        baseName: "SimplifyListLabels",
        type: "boolean",
    }
];
