// asposewordscloud/dist/model/txtSaveOptionsBaseData.js
"use strict";
/*
 * --------------------------------------------------------------------------------
 * <copyright company="Aspose" file="txtSaveOptionsBaseData.ts">
 *   Copyright (c) 2024 Aspose.Words for Cloud
 * </copyright>
 * <summary>
 *   Permission is hereby granted, free of charge, to any person obtaining a copy
 *  of this software and associated documentation files (the "Software"), to deal
 *  in the Software without restriction, including without limitation the rights
 *  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 *  copies of the Software, and to permit persons to whom the Software is
 *  furnished to do so, subject to the following conditions:
 *
 *  The above copyright notice and this permission notice shall be included in all
 *  copies or substantial portions of the Software.
 *
 *  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 *  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 *  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 *  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 *  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 *  SOFTWARE.
 * </summary>
 * --------------------------------------------------------------------------------
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TxtSaveOptionsBaseData = exports.importsMapTxtSaveOptionsBaseData = void 0;
const saveOptionsData_1 = require("./saveOptionsData");
exports.importsMapTxtSaveOptionsBaseData = {
    SaveOptionsData: saveOptionsData_1.SaveOptionsData,
};
/**
 * Base class for save options of text formats.
 */
class TxtSaveOptionsBaseData extends saveOptionsData_1.SaveOptionsData {
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return super.getAttributeTypeMap().concat(TxtSaveOptionsBaseData.attributeTypeMap);
    }
    constructor(init) {
        super(init);
        Object.assign(this, init);
    }
    collectFilesContent(_resultFilesContent) {
    }
    validate() {
        super.validate();
    }
}
exports.TxtSaveOptionsBaseData = TxtSaveOptionsBaseData;
/**
 * Attribute type map
 */
TxtSaveOptionsBaseData.attributeTypeMap = [
    {
        name: "encoding",
        baseName: "Encoding",
        type: "string",
    },
    {
        name: "exportHeadersFootersMode",
        baseName: "ExportHeadersFootersMode",
        type: "TxtSaveOptionsBaseData.ExportHeadersFootersModeEnum",
    },
    {
        name: "forcePageBreaks",
        baseName: "ForcePageBreaks",
        type: "boolean",
    },
    {
        name: "paragraphBreak",
        baseName: "ParagraphBreak",
        type: "string",
    }
];
/**
 * Enums for TxtSaveOptionsBaseData
 */
// tslint:disable:quotemark
// tslint:disable-next-line:no-namespace
(function (TxtSaveOptionsBaseData) {
    let ExportHeadersFootersModeEnum;
    (function (ExportHeadersFootersModeEnum) {
        ExportHeadersFootersModeEnum[ExportHeadersFootersModeEnum["None"] = 'None'] = "None";
        ExportHeadersFootersModeEnum[ExportHeadersFootersModeEnum["PrimaryOnly"] = 'PrimaryOnly'] = "PrimaryOnly";
        ExportHeadersFootersModeEnum[ExportHeadersFootersModeEnum["AllAtEnd"] = 'AllAtEnd'] = "AllAtEnd";
    })(ExportHeadersFootersModeEnum = TxtSaveOptionsBaseData.ExportHeadersFootersModeEnum || (TxtSaveOptionsBaseData.ExportHeadersFootersModeEnum = {}));
})(TxtSaveOptionsBaseData = exports.TxtSaveOptionsBaseData || (exports.TxtSaveOptionsBaseData = {}));
// tslint:enable:quotemark
