// asposewordscloud/dist/model/watermarkDataImage.js
"use strict";
/*
 * --------------------------------------------------------------------------------
 * <copyright company="Aspose" file="watermarkDataImage.ts">
 *   Copyright (c) 2024 Aspose.Words for Cloud
 * </copyright>
 * <summary>
 *   Permission is hereby granted, free of charge, to any person obtaining a copy
 *  of this software and associated documentation files (the "Software"), to deal
 *  in the Software without restriction, including without limitation the rights
 *  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 *  copies of the Software, and to permit persons to whom the Software is
 *  furnished to do so, subject to the following conditions:
 *
 *  The above copyright notice and this permission notice shall be included in all
 *  copies or substantial portions of the Software.
 *
 *  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 *  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 *  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 *  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 *  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 *  SOFTWARE.
 * </summary>
 * --------------------------------------------------------------------------------
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.WatermarkDataImage = exports.importsMapWatermarkDataImage = void 0;
const fileReference_1 = require("./fileReference");
const watermarkDataBase_1 = require("./watermarkDataBase");
exports.importsMapWatermarkDataImage = {
    FileReference: fileReference_1.FileReference,
    WatermarkDataBase: watermarkDataBase_1.WatermarkDataBase,
};
/**
 * Class for insert watermark image request building.
 */
class WatermarkDataImage extends watermarkDataBase_1.WatermarkDataBase {
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return super.getAttributeTypeMap().concat(WatermarkDataImage.attributeTypeMap);
    }
    constructor(init) {
        super(init);
        Object.assign(this, init);
    }
    collectFilesContent(_resultFilesContent) {
        super.collectFilesContent(_resultFilesContent);
        if (this.image) {
            this.image.collectFilesContent(_resultFilesContent);
        }
    }
    validate() {
        var _a;
        super.validate();
        if (this.image === null || this.image === undefined) {
            throw new Error('Property Image in WatermarkDataImage is required.');
        }
        (_a = this.image) === null || _a === void 0 ? void 0 : _a.validate();
    }
}
exports.WatermarkDataImage = WatermarkDataImage;
/**
 * Attribute type map
 */
WatermarkDataImage.attributeTypeMap = [
    {
        name: "image",
        baseName: "Image",
        type: "FileReference",
    },
    {
        name: "isWashout",
        baseName: "IsWashout",
        type: "boolean",
    },
    {
        name: "scale",
        baseName: "Scale",
        type: "number",
    }
];
