// asposewordscloud/dist/model/watermarkText.js
"use strict";
/*
 * --------------------------------------------------------------------------------
 * <copyright company="Aspose" file="watermarkText.ts">
 *   Copyright (c) 2024 Aspose.Words for Cloud
 * </copyright>
 * <summary>
 *   Permission is hereby granted, free of charge, to any person obtaining a copy
 *  of this software and associated documentation files (the "Software"), to deal
 *  in the Software without restriction, including without limitation the rights
 *  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 *  copies of the Software, and to permit persons to whom the Software is
 *  furnished to do so, subject to the following conditions:
 *
 *  The above copyright notice and this permission notice shall be included in all
 *  copies or substantial portions of the Software.
 *
 *  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 *  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 *  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 *  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 *  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 *  SOFTWARE.
 * </summary>
 * --------------------------------------------------------------------------------
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.WatermarkText = exports.importsMapWatermarkText = void 0;
exports.importsMapWatermarkText = {};
/**
 * Class for insert watermark text request building.
 * @deprecated This model will be removed in the future.
 */
class WatermarkText {
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return WatermarkText.attributeTypeMap;
    }
    constructor(init) {
        Object.assign(this, init);
    }
    collectFilesContent(_resultFilesContent) {
    }
    validate() {
        if (this.rotationAngle === null || this.rotationAngle === undefined) {
            throw new Error('Property RotationAngle in WatermarkText is required.');
        }
        if (this.text === null || this.text === undefined) {
            throw new Error('Property Text in WatermarkText is required.');
        }
    }
}
exports.WatermarkText = WatermarkText;
/**
 * Attribute type map
 */
WatermarkText.attributeTypeMap = [
    {
        name: "rotationAngle",
        baseName: "RotationAngle",
        type: "number",
    },
    {
        name: "text",
        baseName: "Text",
        type: "string",
    }
];
