// asposewordscloud/dist/model/xpsSaveOptionsData.js
"use strict";
/*
 * --------------------------------------------------------------------------------
 * <copyright company="Aspose" file="xpsSaveOptionsData.ts">
 *   Copyright (c) 2024 Aspose.Words for Cloud
 * </copyright>
 * <summary>
 *   Permission is hereby granted, free of charge, to any person obtaining a copy
 *  of this software and associated documentation files (the "Software"), to deal
 *  in the Software without restriction, including without limitation the rights
 *  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 *  copies of the Software, and to permit persons to whom the Software is
 *  furnished to do so, subject to the following conditions:
 *
 *  The above copyright notice and this permission notice shall be included in all
 *  copies or substantial portions of the Software.
 *
 *  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 *  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 *  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 *  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 *  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 *  SOFTWARE.
 * </summary>
 * --------------------------------------------------------------------------------
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.XpsSaveOptionsData = exports.importsMapXpsSaveOptionsData = void 0;
const digitalSignatureDetails_1 = require("./digitalSignatureDetails");
const fixedPageSaveOptionsData_1 = require("./fixedPageSaveOptionsData");
const outlineOptionsData_1 = require("./outlineOptionsData");
exports.importsMapXpsSaveOptionsData = {
    DigitalSignatureDetails: digitalSignatureDetails_1.DigitalSignatureDetails,
    FixedPageSaveOptionsData: fixedPageSaveOptionsData_1.FixedPageSaveOptionsData,
    OutlineOptionsData: outlineOptionsData_1.OutlineOptionsData,
};
/**
 * Container class for xps save options.
 */
class XpsSaveOptionsData extends fixedPageSaveOptionsData_1.FixedPageSaveOptionsData {
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return super.getAttributeTypeMap().concat(XpsSaveOptionsData.attributeTypeMap);
    }
    constructor(init) {
        super(init);
        this.saveFormat = 'xps';
        Object.assign(this, init);
    }
    collectFilesContent(_resultFilesContent) {
    }
    validate() {
        var _a, _b;
        super.validate();
        (_a = this.digitalSignatureDetails) === null || _a === void 0 ? void 0 : _a.validate();
        (_b = this.outlineOptions) === null || _b === void 0 ? void 0 : _b.validate();
    }
}
exports.XpsSaveOptionsData = XpsSaveOptionsData;
/**
 * Attribute type map
 */
XpsSaveOptionsData.attributeTypeMap = [
    {
        name: "bookmarksOutlineLevel",
        baseName: "BookmarksOutlineLevel",
        type: "number",
    },
    {
        name: "digitalSignatureDetails",
        baseName: "DigitalSignatureDetails",
        type: "DigitalSignatureDetails",
    },
    {
        name: "headingsOutlineLevels",
        baseName: "HeadingsOutlineLevels",
        type: "number",
    },
    {
        name: "outlineOptions",
        baseName: "OutlineOptions",
        type: "OutlineOptionsData",
    },
    {
        name: "useBookFoldPrintingSettings",
        baseName: "UseBookFoldPrintingSettings",
        type: "boolean",
    }
];
